src/pr-trigger.ts

```diff
@@ -4,175 +4,147 @@ import { queryPRInfo, deriveStateForPR } from "./pr-info";
-        const repo = webhook.repository.name;
-        const sha = webhook.check_suite.head_sha;
-        const pr = await runQueryToGetPRMetadataForSHA1(owner, repo, sha);
-        if (!pr || pr.closed) {
-            const whatFailed = !pr ? "a PR" : "an open PR";
-            context.log.info(`Skipped webhook, could not find ${whatFailed} for the sha referenced in the status (${sha})`);
+    const eventHandler = createEventHandler({});
+    eventHandler.on([
+        "check_suite.completed",
+        "issue_comment.created",
+        "issue_comment.deleted",
+        "issue_comment.edited",
+        "project_card.moved",
+        "pull_request.closed",
+        "pull_request.edited",
+        "pull_request.opened",
+        "pull_request.ready_for_review",
+        "pull_request.reopened",
+        "pull_request.synchronize",
+        "pull_request_review.dismissed",
+        "pull_request_review.submitted",
+    ], async event => {
+        if (event.payload.sender.login === "typescript-bot") {
+            context.log.info(`Skipped webhook because it was triggered by typescript-bot`);
             context.res = {
                 status: 204,
-                body: `NOOPing due to not finding ${whatFailed} for the sha ${sha}`
+                body: `NOOPing because typescript-bot triggered the request`
             };
             return;
         }
-        prNumber = pr.number;
-    }
-
-    if (prNumber === -1) throw new Error(`PR Number was not set from a webhook - ${event} on ${action}`);
 
-    if (prHandlers.has(prNumber)) prHandlers.get(prNumber)(); // cancel older handler for the same pr
-    const aborted = await new Promise(res => {
-        const timeout = setTimeout(() => res(false), 30000);
-        prHandlers.set(prNumber, () => {
-            clearTimeout(timeout);
-            res(true);
-        });
-    });
-    if (aborted) {
-        context.log.info(`Skipped webhook, superseded by a newer one for ${prNumber}`);
-        context.res = {
-            status: 204,
-            body: "NOOPing due to a newer webhook"
-        };
-        return;
-    }
-    prHandlers.delete(prNumber);
-
-    context.log.info(`Getting info for PR ${prNumber} - ${prTitle}`);
-
-    // Generate the info for the PR from scratch
-    const info = await queryPRInfo(prNumber);
-    const prInfo = info.data.repository?.pullRequest;
+        // Allow the bot to run side-effects which are not the 'core' function
+        // of the review cycle, but are related to keeping DT running smoothly
+        if (event.name === "check_suite") {
+            await mergeCodeOwnersOnGreen(event.payload);
+        }
 
-    // If it didn't work, bail early
-    if (!prInfo) {
-        const isIssueNotPR = "issue" in webhook;
-        if (isIssueNotPR) {
+        const pr: { number: number, title?: string } | undefined = prFromEvent();
+        if (!pr) {
+            context.log.info("Skipped webhook, could not find a PR");
```

Comment:
1. The previous code threw an error here, which would make discovering such failures faster
2. It also printed the event+action, which would be useful to debug in case it fails

Reply:
Done, I restored the exception from the previous code. :heavy_check_mark:

Reply:
👍